columnflow/config_util.py

```diff
@@ -83,6 +84,45 @@ def get_shifts_from_sources(config: od.Config, *shift_sources: str) -> list[od.S
     )
 
 
+def create_category_id(
+    config: od.Config,
+    category_name: str,
+    hash_len: int = 8,
+    salt: Any = None,
+) -> int:
+    """
+    Creates a unique id for a :py:class:`order.Category` named *category_name* in a
+    :py:class:`order.Config` object *config* and returns it. Internally,
+    :py:func:`law.util.create_hash` is used which receives *hash_len*. In case of an unintentional
+    (yet unlikely) collision of two ids, there is the option to add a custom *salt* value.
+    """
+    # create the hash
+    h = law.util.create_hash((config.name, config.id, category_name, salt), l=hash_len)
+    h = int(h, base=16)
+
+    # take the last number and add it to the front to ensure that are hashes are above a threshold
+    digits = len(str(int("F" * hash_len, base=16)))
+    h += max(h % 10, 1) * 10 ** digits
```

Comment:
I'm a bit confused by the offset you chose here. Why do you need different offsets based on the last digit of the category?
I'd simply scale each category by the same fixed value, e.g.
```suggestion
    h += 10 ** (digits + 1)
```

Reply:
True, we can do that too (the `+ 1` should be dropped though).

Reply:
Yeah, skipping the `+ 1` is fine with me